### ISSUE TITLE
CUxD 16 channel universal device (40) only 1 channel usable

### ISSUE THREAD

**Issue by KoMa1978** [openhab/openhab-addons#10270]
Hello all,

I'm setting up openHAB to see if it makes sense to switch from iobroker.

At the moment I'm facing an issue that my CUxD 16 channel universal device (type = 40) is discovered correctly in openHAB but only with 1 channel. The other 15 channels are not discovered.

It is possible to configure this device in CUxD to behave like a switch or button or whatever and openHAB discovers it as "HM-LC-Sw1PBU-FM" which (in general) has 1 channel. But in this setup it has more channels and this is working fine in HomeMatic and also in iobroker.

In case the code is needed:

`UID: homematic:HM-LC-Sw1PBU-FM:xxxxxxx:CUX4000002
label: FB_Status
thingTypeUID: homematic:HM-LC-Sw1PBU-FM
configuration:
  HMP_1_STATUSINFO_MINDELAY: 2
  HMP_13_CMD_EXEC: false
  HMP_4_CMD_EXEC: false
  HMP_7_REG_MATCH: false
  HMP_1_AES_ACTIVE: false
  HMP_9_REG_MATCH: false
  HMP_14_CMD_EXEC: false
  HMP_3_CMD_EXEC: false
  HMP_-1_CONTROL: SWITCH
  HMP_16_REG_MATCH: false
  HMP_3_ACTIVE: true
  HMP_12_REPEAT: 0
  HMP_13_REPEAT: 0
  HMP_14_ACTIVE: true
  HMP_13_ACTIVE: true
  HMP_16_REPEAT: 0
  HMP_7_ACTIVE: true
  HMP_6_REG_MATCH: false
  HMP_14_REG_MATCH: false
  HMP_8_ACTIVE: true
  HMP_1_STATUSINFO_RANDOM: 1
  HMP_4_ACTIVE: true
  HMP_10_ACTIVE: true
  HMP_-1_LOCAL_RESET_DISABLE: false
  HMP_11_REG_MATCH: false
  HMP_5_CMD_EXEC: false
  HMP_1_REG_MATCH: false
  HMP_2_REPEAT: 0
  HMP_6_REPEAT: 0
  HMP_1_CMD_EXEC: false
  HMP_4_REG_MATCH: false
  HMP_1_TRANSMIT_TRY_MAX: 6
  HMP_11_CMD_EXEC: false
  HMP_7_CMD_EXEC: false
  HMP_16_CMD_EXEC: false
  HMP_12_REG_MATCH: false
  HMP_8_CMD_EXEC: false
  HMP_13_REG_MATCH: false
  HMP_2_CMD_EXEC: false
  HMP_10_CMD_EXEC: false
  HMP_3_REG_MATCH: false
  HMP_3_REPEAT: 0
  HMP_1_REPEAT: 0
  HMP_-1_SYSLOG: false
  HMP_5_REPEAT: 0
  HMP_7_REPEAT: 0
  HMP_9_REPEAT: 0
  HMP_12_CMD_EXEC: false
  HMP_15_CMD_EXEC: false
  HMP_16_ACTIVE: true
  HMP_2_ACTIVE: true
  HMP_10_REPEAT: 0
  HMP_11_REPEAT: 0
  HMP_14_REPEAT: 0
  HMP_15_ACTIVE: true
  HMP_10_REG_MATCH: false
  HMP_6_ACTIVE: true
  HMP_15_REPEAT: 0
  HMP_1_ACTIVE: true
  HMP_5_ACTIVE: true
  HMP_9_CMD_EXEC: false
  HMP_9_ACTIVE: true
  HMP_11_ACTIVE: true
  HMP_12_ACTIVE: true
  HMP_2_REG_MATCH: false
  HMP_-1_INTERNAL_KEYS_VISIBLE: true
  HMP_15_REG_MATCH: false
  HMP_5_REG_MATCH: false
  HMP_6_CMD_EXEC: false
  HMP_1_POWERUP_ACTION: "0"
  HMP_8_REG_MATCH: false
  HMP_4_REPEAT: 0
  HMP_8_REPEAT: 0
bridgeUID: homematic:bridge:xxxxxxx`

Adding it in Expert Mode doesn't help, it adds only another channel which controls only channel number 1:
`Switch    FBStatus_State                   "State"                    <Switch>            ["Point"]  { channel="homematic:HM-LC-Sw1PBU-FM:xxx:CUX4000002:1#STATE" }                
Switch    FBStatus_State                   "State"                    <Switch>            ["Point"]  { channel="homematic:HM-LC-Sw1PBU-FM:xxx:CUX4000002:2#STATE" }`

Any idea how I can solve this?

KoMa1978

**Comment by MHerbst** [comment]
The binding creates the channels based on information it receives from Homematic. Did you check whether you can see the channels if you select "Show advanced"?

If not we need some additional information

- type and version of the CCU
- CuxD version
- openHAB version

As a next step, please delete the thing in OH, set the log level to TRACE (`log:set TRACE org.openhab.binding.homematic`), start a new discovery and post the openhab.log with the information starting from the discovery process.

**Comment by KoMa1978** [comment]
Hi,

yes I checked Advanced but the channels are not available. I assume this is more or less related to the fact that the CUxD device is emulated as a switch "HM-LC-Sw1PBU-FM" which normally has only one channel.

I had a similar issue with JARVIS, a frontend plugin for iobroker, while iobroker itself handles the device correctly (https://github.com/Zefau/ioBroker.jarvis/issues/442 in German language but perhaps it helps).

openHAB: 3.01
CUxD: 2.5
HomeMatic: 3.55.10

It looks like the channels are discovered correctly but only not displayed/used ...

If you need more info or tests, I'm sure I can provide.
[trace01.txt](https://github.com/openhab/openhab-addons/files/6073017/trace01.txt)
[trace02.txt](https://github.com/openhab/openhab-addons/files/6073018/trace02.txt)

**Comment by KoMa1978** [comment]
I upgraded Homematic to 3.57.4 and CUxD to 2.5.1 and did a rescan again. Same issue.

I have imported another CUXD Device which is identical to the device I used for this issue: Same ... only state of channel 1 is available.

**Comment by MHerbst** [comment]
The log does not really help. I forgot to mention that you have to remove all devices of the same time. Otherwise I can see only messages about cloning a device.
Can you please try it again but remove all devices of this type before you try the discovery again. Please send the log file in one file. To me it looks like some relevant information may be missing.

**Comment by KoMa1978** [comment]
I don't understand ... I removed other devices from the log and grabbed only the information from the specific device and just don't want to upload the whole 15MB file because of all the sensitive data like serial number etc.

And if I look at both logs, I see that some information regarding the other 15 channels are discovered and "Channel 1" and "Channel 2" (and the other channels) are looking very similar. It seems that all the information is discovered correctly but not intepreted correctly. I assume because openhab is "confused" because it is a "HM-LC-Sw1PBU-FM" which normally has only 1 channel.

I still have the whole log. What exactly are you looking for.

Below are some examples why I think the discovery was OK (excerpt from the uploaded logs):
```
		ADDRESS=CUX4000002
		AES_ACTIVE=0
		CHILDREN=
		[
			CUX4000002:0
			CUX4000002:1
			CUX4000002:2
			CUX4000002:3
			CUX4000002:4
			CUX4000002:5
			CUX4000002:6
			CUX4000002:7
			CUX4000002:8
			CUX4000002:9
			CUX4000002:10
			CUX4000002:11
			CUX4000002:12
			CUX4000002:13
			CUX4000002:14
			CUX4000002:15
			CUX4000002:16
```

```
	{
		ADDRESS=CUX4000002:1
		AES_ACTIVE=0
		DIRECTION=0
		FLAGS=1
		INDEX=1
		LINK_SOURCE_ROLES=
		LINK_TARGET_ROLES=
		PARAMSETS=
		[
			MASTER
			VALUES
		]
		PARENT=CUX4000002
		PARENT_TYPE=HM-LC-Sw1PBU-FM
		TYPE=SWITCH
		VERSION=1
	}
	{
		ADDRESS=CUX4000002:2
		AES_ACTIVE=0
		DIRECTION=0
		FLAGS=1
		INDEX=2
		LINK_SOURCE_ROLES=
		LINK_TARGET_ROLES=
		PARAMSETS=
		[
			MASTER
			VALUES
		]
		PARENT=CUX4000002
		PARENT_TYPE=HM-LC-Sw1PBU-FM
		TYPE=SWITCH
		VERSION=1
	}
```

same for the other log file:

```
2021-03-03 02:20:36.438 [TRACE] [ommunicator.AbstractHomematicGateway] -   HmChannel[number=1,type=SWITCH,initialized=false]
2021-03-03 02:20:36.438 [TRACE] [ommunicator.AbstractHomematicGateway] -     HmDatapoint[name=CONTROL,value=<null>,defaultValue=0,type=INTEGER,minValue=0,maxValue=8,step=<null>,options=<null>,readOnly=true,readable=true,unit=<null>,description=CONTROL,info=<null>,paramsetType=VALUES,virtual=false,trigger=false]
2021-03-03 02:20:36.438 [TRACE] [ommunicator.AbstractHomematicGateway] -     HmDatapoint[name=PRESS_SHORT,value=<null>,defaultValue=false,type=ACTION,minValue=<null>,maxValue=<null>,step=<null>,options=<null>,readOnly=false,readable=false,unit=<null>,description=PRESS_SHORT,info=<null>,paramsetType=VALUES,virtual=false,trigger=false]
2021-03-03 02:20:36.438 [TRACE] [ommunicator.AbstractHomematicGateway] -     HmDatapoint[name=RCV_SHORT,value=<null>,defaultValue=<null>,type=STRING,minValue=<null>,maxValue=<null>,step=<null>,options=<null>,readOnly=false,readable=true,unit=<null>,description=RCV_SHORT,info=<null>,paramsetType=MASTER,virtual=false,trigger=false]
2021-03-03 02:20:36.438 [TRACE] [ommunicator.AbstractHomematicGateway] -     HmDatapoint[name=LEVEL,value=<null>,defaultValue=0.0,type=FLOAT,minValue=0.0,maxValue=1.0,step=<null>,options=<null>,readOnly=false,readable=true,unit=100%,description=LEVEL,info=<null>,paramsetType=VALUES,virtual=false,trigger=false]
2021-03-03 02:20:36.438 [TRACE] [ommunicator.AbstractHomematicGateway] -     HmDatapoint[name=CMD_SHORT,value=<null>,defaultValue=<null>,type=STRING,minValue=<null>,maxValue=<null>,step=<null>,options=<null>,readOnly=false,readable=true,unit=<null>,description=CMD_SHORT,info=<null>,paramsetType=MASTER,virtual=false,trigger=false]
2021-03-03 02:20:36.438 [TRACE] [ommunicator.AbstractHomematicGateway] -     HmDatapoint[name=PRESS_LONG,value=<null>,defaultValue=false,type=ACTION,minValue=<null>,maxValue=<null>,step=<null>,options=<null>,readOnly=false,readable=false,unit=<null>,description=PRESS_LONG,info=<null>,paramsetType=VALUES,virtual=false,trigger=false]
2021-03-03 02:20:36.438 [TRACE] [ommunicator.AbstractHomematicGateway] -     HmDatapoint[name=TOGGLE,value=<null>,defaultValue=false,type=ACTION,minValue=<null>,maxValue=<null>,step=<null>,options=<null>,readOnly=false,readable=false,unit=<null>,description=TOGGLE,info=<null>,paramsetType=VALUES,virtual=false,trigger=false]
2021-03-03 02:20:36.438 [TRACE] [ommunicator.AbstractHomematicGateway] -     HmDatapoint[name=RCVL,value=<null>,defaultValue=<null>,type=STRING,minValue=<null>,maxValue=<null>,step=<null>,options=<null>,readOnly=true,readable=true,unit=<null>,description=RCVL,info=<null>,paramsetType=VALUES,virtual=false,trigger=false]
2021-03-03 02:20:36.438 [TRACE] [ommunicator.AbstractHomematicGateway] -     HmDatapoint[name=CMD_EXEC,value=<null>,defaultValue=false,type=BOOL,minValue=<null>,maxValue=<null>,step=<null>,options=<null>,readOnly=false,readable=true,unit=<null>,description=CMD_EXEC,info=<null>,paramsetType=MASTER,virtual=false,trigger=false]
2021-03-03 02:20:36.438 [TRACE] [ommunicator.AbstractHomematicGateway] -     HmDatapoint[name=REG_MATCH,value=<null>,defaultValue=false,type=BOOL,minValue=<null>,maxValue=<null>,step=<null>,options=<null>,readOnly=false,readable=true,unit=<null>,description=REG_MATCH,info=<null>,paramsetType=MASTER,virtual=false,trigger=false]
2021-03-03 02:20:36.438 [TRACE] [ommunicator.AbstractHomematicGateway] -     HmDatapoint[name=ACTIVE,value=<null>,defaultValue=true,type=BOOL,minValue=<null>,maxValue=<null>,step=<null>,options=<null>,readOnly=false,readable=true,unit=<null>,description=ACTIVE,info=<null>,paramsetType=MASTER,virtual=false,trigger=false]
2021-03-03 02:20:36.439 [TRACE] [ommunicator.AbstractHomematicGateway] -     HmDatapoint[name=BUTTON,value=<null>,defaultValue=<null>,type=STRING,minValue=<null>,maxValue=<null>,step=<null>,options=SHORT_PRESSED;LONG_PRESSED;DOUBLE_PRESSED,readOnly=false,readable=true,unit=<null>,description=BUTTON,info=<null>,paramsetType=VALUES,virtual=true,trigger=true]
2021-03-03 02:20:36.439 [TRACE] [ommunicator.AbstractHomematicGateway] -     HmDatapoint[name=STATE,value=<null>,defaultValue=false,type=BOOL,minValue=<null>,maxValue=<null>,step=<null>,options=<null>,readOnly=false,readable=true,unit=<null>,description=STATE,info=<null>,paramsetType=VALUES,virtual=false,trigger=false]
2021-03-03 02:20:36.439 [TRACE] [ommunicator.AbstractHomematicGateway] -     HmDatapoint[name=INSTALL_TEST,value=<null>,defaultValue=false,type=ACTION,minValue=<null>,maxValue=<null>,step=<null>,options=<null>,readOnly=false,readable=false,unit=<null>,description=INSTALL_TEST,info=<null>,paramsetType=VALUES,virtual=false,trigger=false]
2021-03-03 02:20:36.439 [TRACE] [ommunicator.AbstractHomematicGateway] -     HmDatapoint[name=STOP,value=<null>,defaultValue=false,type=ACTION,minValue=<null>,maxValue=<null>,step=<null>,options=<null>,readOnly=false,readable=false,unit=<null>,description=STOP,info=<null>,paramsetType=VALUES,virtual=false,trigger=false]
2021-03-03 02:20:36.439 [TRACE] [ommunicator.AbstractHomematicGateway] -     HmDatapoint[name=RCVS,value=<null>,defaultValue=<null>,type=STRING,minValue=<null>,maxValue=<null>,step=<null>,options=<null>,readOnly=true,readable=true,unit=<null>,description=RCVS,info=<null>,paramsetType=VALUES,virtual=false,trigger=false]
2021-03-03 02:20:36.439 [TRACE] [ommunicator.AbstractHomematicGateway] -     HmDatapoint[name=DEVICE,value=<null>,defaultValue=<null>,type=STRING,minValue=<null>,maxValue=<null>,step=<null>,options=<null>,readOnly=false,readable=true,unit=<null>,description=DEVICE,info=<null>,paramsetType=MASTER,virtual=false,trigger=false]
2021-03-03 02:20:36.439 [TRACE] [ommunicator.AbstractHomematicGateway] -     HmDatapoint[name=MOTION,value=<null>,defaultValue=false,type=BOOL,minValue=<null>,maxValue=<null>,step=<null>,options=<null>,readOnly=true,readable=true,unit=<null>,description=MOTION,info=<null>,paramsetType=VALUES,virtual=false,trigger=false]
2021-03-03 02:20:36.439 [TRACE] [ommunicator.AbstractHomematicGateway] -     HmDatapoint[name=OLD_LEVEL,value=<null>,defaultValue=false,type=ACTION,minValue=<null>,maxValue=<null>,step=<null>,options=<null>,readOnly=false,readable=false,unit=<null>,description=OLD_LEVEL,info=<null>,paramsetType=VALUES,virtual=false,trigger=false]
2021-03-03 02:20:36.439 [TRACE] [ommunicator.AbstractHomematicGateway] -     HmDatapoint[name=RCV_LONG,value=<null>,defaultValue=<null>,type=STRING,minValue=<null>,maxValue=<null>,step=<null>,options=<null>,readOnly=false,readable=true,unit=<null>,description=RCV_LONG,info=<null>,paramsetType=MASTER,virtual=false,trigger=false]
2021-03-03 02:20:36.439 [TRACE] [ommunicator.AbstractHomematicGateway] -     HmDatapoint[name=SET_STATE,value=<null>,defaultValue=0.0,type=FLOAT,minValue=0.0,maxValue=2.0,step=<null>,options=<null>,readOnly=false,readable=false,unit=<null>,description=SET_STATE,info=<null>,paramsetType=VALUES,virtual=false,trigger=false]
2021-03-03 02:20:36.439 [TRACE] [ommunicator.AbstractHomematicGateway] -     HmDatapoint[name=CMD_LONG,value=<null>,defaultValue=<null>,type=STRING,minValue=<null>,maxValue=<null>,step=<null>,options=<null>,readOnly=false,readable=true,unit=<null>,description=CMD_LONG,info=<null>,paramsetType=MASTER,virtual=false,trigger=false]
2021-03-03 02:20:36.439 [TRACE] [ommunicator.AbstractHomematicGateway] -     HmDatapoint[name=SEND_CMD,value=<null>,defaultValue=<null>,type=STRING,minValue=<null>,maxValue=<null>,step=<null>,options=<null>,readOnly=false,readable=false,unit=<null>,description=SEND_CMD,info=<null>,paramsetType=VALUES,virtual=false,trigger=false]
2021-03-03 02:20:36.439 [TRACE] [ommunicator.AbstractHomematicGateway] -     HmDatapoint[name=REPEAT,value=<null>,defaultValue=0,type=INTEGER,minValue=0,maxValue=2,step=<null>,options=<null>,readOnly=false,readable=true,unit=<null>,description=REPEAT,info=<null>,paramsetType=MASTER,virtual=false,trigger=false]
2021-03-03 02:20:36.439 [TRACE] [ommunicator.AbstractHomematicGateway] -   HmChannel[number=2,type=SWITCH,initialized=false]
2021-03-03 02:20:36.439 [TRACE] [ommunicator.AbstractHomematicGateway] -     HmDatapoint[name=LEVEL,value=<null>,defaultValue=0.0,type=FLOAT,minValue=0.0,maxValue=1.0,step=<null>,options=<null>,readOnly=false,readable=true,unit=100%,description=LEVEL,info=<null>,paramsetType=VALUES,virtual=false,trigger=false]
2021-03-03 02:20:36.439 [TRACE] [ommunicator.AbstractHomematicGateway] -     HmDatapoint[name=REPEAT,value=<null>,defaultValue=0,type=INTEGER,minValue=0,maxValue=2,step=<null>,options=<null>,readOnly=false,readable=true,unit=<null>,description=REPEAT,info=<null>,paramsetType=MASTER,virtual=false,trigger=false]
2021-03-03 02:20:36.439 [TRACE] [ommunicator.AbstractHomematicGateway] -     HmDatapoint[name=CMD_SHORT,value=<null>,defaultValue=<null>,type=STRING,minValue=<null>,maxValue=<null>,step=<null>,options=<null>,readOnly=false,readable=true,unit=<null>,description=CMD_SHORT,info=<null>,paramsetType=MASTER,virtual=false,trigger=false]
2021-03-03 02:20:36.439 [TRACE] [ommunicator.AbstractHomematicGateway] -     HmDatapoint[name=MOTION,value=<null>,defaultValue=false,type=BOOL,minValue=<null>,maxValue=<null>,step=<null>,options=<null>,readOnly=true,readable=true,unit=<null>,description=MOTION,info=<null>,paramsetType=VALUES,virtual=false,trigger=false]
2021-03-03 02:20:36.439 [TRACE] [ommunicator.AbstractHomematicGateway] -     HmDatapoint[name=STATE,value=<null>,defaultValue=false,type=BOOL,minValue=<null>,maxValue=<null>,step=<null>,options=<null>,readOnly=false,readable=true,unit=<null>,description=STATE,info=<null>,paramsetType=VALUES,virtual=false,trigger=false]
2021-03-03 02:20:36.439 [TRACE] [ommunicator.AbstractHomematicGateway] -     HmDatapoint[name=BUTTON,value=<null>,defaultValue=<null>,type=STRING,minValue=<null>,maxValue=<null>,step=<null>,options=SHORT_PRESSED;LONG_PRESSED;DOUBLE_PRESSED,readOnly=false,readable=true,unit=<null>,description=BUTTON,info=<null>,paramsetType=VALUES,virtual=true,trigger=true]
2021-03-03 02:20:36.439 [TRACE] [ommunicator.AbstractHomematicGateway] -     HmDatapoint[name=ACTIVE,value=<null>,defaultValue=true,type=BOOL,minValue=<null>,maxValue=<null>,step=<null>,options=<null>,readOnly=false,readable=true,unit=<null>,description=ACTIVE,info=<null>,paramsetType=MASTER,virtual=false,trigger=false]
2021-03-03 02:20:36.439 [TRACE] [ommunicator.AbstractHomematicGateway] -     HmDatapoint[name=STOP,value=<null>,defaultValue=false,type=ACTION,minValue=<null>,maxValue=<null>,step=<null>,options=<null>,readOnly=false,readable=false,unit=<null>,description=STOP,info=<null>,paramsetType=VALUES,virtual=false,trigger=false]
2021-03-03 02:20:36.439 [TRACE] [ommunicator.AbstractHomematicGateway] -     HmDatapoint[name=CONTROL,value=<null>,defaultValue=0,type=INTEGER,minValue=0,maxValue=8,step=<null>,options=<null>,readOnly=true,readable=true,unit=<null>,description=CONTROL,info=<null>,paramsetType=VALUES,virtual=false,trigger=false]
2021-03-03 02:20:36.439 [TRACE] [ommunicator.AbstractHomematicGateway] -     HmDatapoint[name=CMD_EXEC,value=<null>,defaultValue=false,type=BOOL,minValue=<null>,maxValue=<null>,step=<null>,options=<null>,readOnly=false,readable=true,unit=<null>,description=CMD_EXEC,info=<null>,paramsetType=MASTER,virtual=false,trigger=false]
2021-03-03 02:20:36.439 [TRACE] [ommunicator.AbstractHomematicGateway] -     HmDatapoint[name=PRESS_SHORT,value=<null>,defaultValue=false,type=ACTION,minValue=<null>,maxValue=<null>,step=<null>,options=<null>,readOnly=false,readable=false,unit=<null>,description=PRESS_SHORT,info=<null>,paramsetType=VALUES,virtual=false,trigger=false]
2021-03-03 02:20:36.439 [TRACE] [ommunicator.AbstractHomematicGateway] -     HmDatapoint[name=RCV_SHORT,value=<null>,defaultValue=<null>,type=STRING,minValue=<null>,maxValue=<null>,step=<null>,options=<null>,readOnly=false,readable=true,unit=<null>,description=RCV_SHORT,info=<null>,paramsetType=MASTER,virtual=false,trigger=false]
2021-03-03 02:20:36.439 [TRACE] [ommunicator.AbstractHomematicGateway] -     HmDatapoint[name=TOGGLE,value=<null>,defaultValue=false,type=ACTION,minValue=<null>,maxValue=<null>,step=<null>,options=<null>,readOnly=false,readable=false,unit=<null>,description=TOGGLE,info=<null>,paramsetType=VALUES,virtual=false,trigger=false]
2021-03-03 02:20:36.440 [TRACE] [ommunicator.AbstractHomematicGateway] -     HmDatapoint[name=CMD_LONG,value=<null>,defaultValue=<null>,type=STRING,minValue=<null>,maxValue=<null>,step=<null>,options=<null>,readOnly=false,readable=true,unit=<null>,description=CMD_LONG,info=<null>,paramsetType=MASTER,virtual=false,trigger=false]
2021-03-03 02:20:36.440 [TRACE] [ommunicator.AbstractHomematicGateway] -     HmDatapoint[name=INSTALL_TEST,value=<null>,defaultValue=false,type=ACTION,minValue=<null>,maxValue=<null>,step=<null>,options=<null>,readOnly=false,readable=false,unit=<null>,description=INSTALL_TEST,info=<null>,paramsetType=VALUES,virtual=false,trigger=false]
2021-03-03 02:20:36.440 [TRACE] [ommunicator.AbstractHomematicGateway] -     HmDatapoint[name=REG_MATCH,value=<null>,defaultValue=false,type=BOOL,minValue=<null>,maxValue=<null>,step=<null>,options=<null>,readOnly=false,readable=true,unit=<null>,description=REG_MATCH,info=<null>,paramsetType=MASTER,virtual=false,trigger=false]
2021-03-03 02:20:36.440 [TRACE] [ommunicator.AbstractHomematicGateway] -     HmDatapoint[name=RCVL,value=<null>,defaultValue=<null>,type=STRING,minValue=<null>,maxValue=<null>,step=<null>,options=<null>,readOnly=true,readable=true,unit=<null>,description=RCVL,info=<null>,paramsetType=VALUES,virtual=false,trigger=false]
2021-03-03 02:20:36.440 [TRACE] [ommunicator.AbstractHomematicGateway] -     HmDatapoint[name=PRESS_LONG,value=<null>,defaultValue=false,type=ACTION,minValue=<null>,maxValue=<null>,step=<null>,options=<null>,readOnly=false,readable=false,unit=<null>,description=PRESS_LONG,info=<null>,paramsetType=VALUES,virtual=false,trigger=false]
2021-03-03 02:20:36.440 [TRACE] [ommunicator.AbstractHomematicGateway] -     HmDatapoint[name=SEND_CMD,value=<null>,defaultValue=<null>,type=STRING,minValue=<null>,maxValue=<null>,step=<null>,options=<null>,readOnly=false,readable=false,unit=<null>,description=SEND_CMD,info=<null>,paramsetType=VALUES,virtual=false,trigger=false]
2021-03-03 02:20:36.440 [TRACE] [ommunicator.AbstractHomematicGateway] -     HmDatapoint[name=OLD_LEVEL,value=<null>,defaultValue=false,type=ACTION,minValue=<null>,maxValue=<null>,step=<null>,options=<null>,readOnly=false,readable=false,unit=<null>,description=OLD_LEVEL,info=<null>,paramsetType=VALUES,virtual=false,trigger=false]
2021-03-03 02:20:36.440 [TRACE] [ommunicator.AbstractHomematicGateway] -     HmDatapoint[name=RCV_LONG,value=<null>,defaultValue=<null>,type=STRING,minValue=<null>,maxValue=<null>,step=<null>,options=<null>,readOnly=false,readable=true,unit=<null>,description=RCV_LONG,info=<null>,paramsetType=MASTER,virtual=false,trigger=false]
2021-03-03 02:20:36.440 [TRACE] [ommunicator.AbstractHomematicGateway] -     HmDatapoint[name=SET_STATE,value=<null>,defaultValue=0.0,type=FLOAT,minValue=0.0,maxValue=2.0,step=<null>,options=<null>,readOnly=false,readable=false,unit=<null>,description=SET_STATE,info=<null>,paramsetType=VALUES,virtual=false,trigger=false]
2021-03-03 02:20:36.440 [TRACE] [ommunicator.AbstractHomematicGateway] -     HmDatapoint[name=DEVICE,value=<null>,defaultValue=<null>,type=STRING,minValue=<null>,maxValue=<null>,step=<null>,options=<null>,readOnly=false,readable=true,unit=<null>,description=DEVICE,info=<null>,paramsetType=MASTER,virtual=false,trigger=false]
2021-03-03 02:20:36.440 [TRACE] [ommunicator.AbstractHomematicGateway] -     HmDatapoint[name=RCVS,value=<null>,defaultValue=<null>,type=STRING,minValue=<null>,maxValue=<null>,step=<null>,options=<null>,readOnly=true,readable=true,unit=<null>,description=RCVS,info=<null>,paramsetType=VALUES,virtual=false,trigger=false]

```

**Comment by KoMa1978** [comment]
Short update ...

If I change the value on one of these switches in HomeMatic, openHAB recognizes it but don't know how to handle it:

```
2021-03-03 19:33:55.882 [WARN ] [ternal.handler.HomematicThingHandler] - Channel not found for datapoint 'CUX4000002:2#STATE'
2021-03-03 19:33:57.885 [WARN ] [ternal.handler.HomematicThingHandler] - Channel not found for datapoint 'CUX4000002:2#STATE'
```
And in trace level:

```
2021-03-03 19:35:47.754 [TRACE] [nicator.server.BinRpcResponseHandler] - Event BinRpcMessage: system.multicall()
[
        {
                methodName=event
                params=
                [
                        CUxD
                        CUX4000002:2
                        STATE
                        true
                ]
        }
]
2021-03-03 19:35:47.754 [DEBUG] [ommunicator.AbstractHomematicGateway] - Received new (Boolean) value 'true' for 'CUX4000002:2#STATE' from gateway with id 'xxx'
2021-03-03 19:35:47.755 [WARN ] [ternal.handler.HomematicThingHandler] - Channel not found for datapoint 'CUX4000002:2#STATE'
2021-03-03 19:35:48.840 [TRACE] [nicator.server.BinRpcResponseHandler] - Event BinRpcMessage: system.multicall()
[
        {
                methodName=event
                params=
                [
                        CUxD
                        CUX4000002:2
                        STATE
                        true
                ]
        }
]
2021-03-03 19:35:48.840 [DEBUG] [ommunicator.AbstractHomematicGateway] - Received new (Boolean) value 'true' for 'CUX4000002:2#STATE' from gateway with id 'xxx'
2021-03-03 19:35:48.840 [WARN ] [ternal.handler.HomematicThingHandler] - Channel not found for datapoint 'CUX4000002:2#STATE'

```

**Comment by MHerbst** [comment]
I asked for the whole log because It is necessary to see it as continuous output. I need especially the information for the first thing that is created. In the attached logs everything looked OK, so it is not clear why not all channels (e.g. STATE) have been created.

In your log excerpt, I saw a message that for the newly created thing the channels have been "cloned". This means it is not the first thing that was created. If the created channels for the first thing are incomplete they will also be incomplete for all other things of the same device type. This means if you also own a real "HM-LC-Sw1PBU-FM" device from Homematic for the CuxD device of the same type you will probably get the same channel definitions.

If you like you can zip the log and send it directly to my mail address.

> I assume because openhab is "confused" because it is a "HM-LC-Sw1PBU-FM" which normally has only 1 channel.

The Homematic binding works differently than other bindings and also differently than Homematic implementations in other Smarthome environments: all information to create things and channels are taken from the information the CCU returns to the binding. Therefore you could give the device a completely different name and type and it would work. 

There are only some special Homematic devices where the automatic derivation of channel definitions does not work and these devices are created from pre-defined configurations stored in the binding. As far as I can see, this is not true for the "HM-LC-Sw1PBU-FM" device type.

I don't know whether you can change the device type for the emulated device in CuxD to something different. If yes, it would be interesting to see whether the binding behaves different.

**Comment by KoMa1978** [comment]
Hi Martin,

first thanks a lot for your support!
After some troubleshooting, we found that the issue occurs because these switches are emulated as "HM-LC-Sw1PBU-FM" and I already have one (yes seriously ... one single ;-) ) real device of type "HM-LC-Sw1PBU-FM".

I created a new device in CuXD of the same type (40 - Universal Device) but this time I chose "Virtual Switch" as device Icon which created a new Homeatic Device of type "VIR-LG-ONOFF" which is now discovered with 16 usuable channels.

This device type behaves similar in HomeMatic, in any App I use and also in any other connected tools (like ioBroker or HomeBridge).

This workaround is fine for me.

Thanks a lot Martin!!
 Marco
